samples/code-samples/ChangeFeedProcessor/DocumentFeedObserver.cs

```diff
@@ -86,19 +86,20 @@ public Task ProcessChangesAsync(ChangeFeedObserverContext context, IReadOnlyList
         {
             Console.ForegroundColor = ConsoleColor.Green;
             Console.WriteLine("Change feed: PartitionId {0} total {1} doc(s)", context.PartitionKeyRangeId, Interlocked.Add(ref totalDocs, docs.Count));
+            var tasks = new List<Task>(docs.Count);
+
             foreach (Document doc in docs)
             {
                 Console.ForegroundColor = ConsoleColor.Yellow;
-                Console.WriteLine(doc.Id.ToString());
-               
+                Console.WriteLine(doc.Id);
 
                 if (this.destinationCollectionUri != null)
                 {
-                    this.client.UpsertDocumentAsync(this.destinationCollectionUri, doc);
+                    tasks.Add(this.client.UpsertDocumentAsync(this.destinationCollectionUri, doc));
```

Comment:
Uncontrolled degree of parallelism is a bit of concern. Using something like ParallelHelper from CFP https://github.com/Azure/azure-documentdb-dotnet/blob/master/samples/ChangeFeedProcessor/DocumentDB.ChangeFeedProcessor/Internal/ParallelHelper.cs would solve it.

Reply:
Original issue was firing tasks in the background and returning, which would not allow to await properly and catch exceptions if those would be thrown. @mkolt I'm not sure this PR is the right place to address it. Rather have a separate PR to control degree of parallelism.

Reply:
@mkolt looking at the helper method you've pointed to, is it efficient to wrap tasks in more tasks just to get partitions? Would it be better to divide already async tasks into chunks/partitions w/o extra Task.Run?

Reply:
@SeanFeldman the original issue was fixed (missing await is there now). I guess, I'll take this back, this is sample, let's keep it simple using adding tasks to a list. WRT Task.Run I think it's good that it makes sure that each task is executed independently without waiting for or blocking other tasks..